src/main/java/com/pinterest/secor/common/ZookeeperConnector.java

```diff
@@ -67,7 +67,7 @@ public ZookeeperConnector(SecorConfig config) {
             assert elements.length == 2: Integer.toString(elements.length) + " == 2";
             String host = elements[0];
             int port = Integer.parseInt(elements[1]);
-            result.add(new InetSocketAddress(host, port));
+            result.add(InetSocketAddress.createUnresolved(host, port));
```

Comment:
I am not quite familiar with this domain, are you using a proxy?  If the address is not resolved, when will it resolve?  And if the ip address of the ZK changed, will this unresolved address re-resolve?